File: Modules/ServerTypes/Announcement/AnnouncementGeneratedEditorForm.ts
import { LookupEditor, StringEditor, MultipleImageUploadEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface AnnouncementGeneratedEditorForm {
    EmployeeRowId: LookupEditor;
    EmployeeName: StringEditor;
    AnnouncementContent: StringEditor;
    UploadDocument: MultipleImageUploadEditor;
}

export class AnnouncementGeneratedEditorForm extends PrefixedContext {
    static formKey = 'Announcement.AnnouncementGeneratedEditor';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!AnnouncementGeneratedEditorForm.init)  {
            AnnouncementGeneratedEditorForm.init = true;

            var w0 = LookupEditor;
            var w1 = StringEditor;
            var w2 = MultipleImageUploadEditor;

            initFormType(AnnouncementGeneratedEditorForm, [
                'EmployeeRowId', w0,
                'EmployeeName', w1,
                'AnnouncementContent', w1,
                'UploadDocument', w2
            ]);
        }
    }
}

File: Modules/HumanResource/Announcement/AnnouncementGenerated/AnnouncementGeneratedEditorForm.cs
using HRMSoftware.Announcement;
using Serenity.ComponentModel;
using System;

namespace HRMSoftware.Announcement.Forms;

[FormScript("Announcement.AnnouncementGeneratedEditor")]
[BasedOnRow(typeof(AnnouncementGeneratedRow), CheckNames = true)]
public class AnnouncementGeneratedEditorForm   
{
    [HalfWidth]
    public int EmployeeRowId { get; set; }
    [HalfWidth]
    public string EmployeeName { get; set; }


    public string AnnouncementContent { get; set; }
    public string UploadDocument { get; set; }
}